ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -201,37 +219,66 @@ export function AssetPickerModal({
       }
 
       for (const token of Object.values(tokenList)) {
-        yield token;
+        if (shouldAddToken(token.symbol, token.address)) {
+          yield token;
+        }
       }
 
       for (const token of blockedTokens) {
         yield token;
       }
-    }
+    },
+    [
+      nativeCurrency,
+      nativeCurrencyImage,
+      balanceValue,
+      memoizedUsersTokens,
+      topTokens,
+      tokenList,
+      getIsDisabled,
+    ],
+  );
 
-    for (const token of tokenGenerator()) {
-      if (
-        token.symbol?.toLowerCase().includes(searchQuery.toLowerCase()) &&
-        !filteredTokensAddresses.has(token.address?.toLowerCase())
-      ) {
-        filteredTokensAddresses.add(token.address?.toLowerCase());
-        filteredTokens.push(
-          getRenderableTokenData(
-            token.address
-              ? ({
-                  ...token,
-                  ...tokenList[token.address.toLowerCase()],
-                  type: AssetType.token,
-                } as AssetWithDisplayData<ERC20Asset>)
-              : token,
-            tokenConversionRates,
-            conversionRate,
-            currentCurrency,
-            chainId,
-            tokenList,
-          ),
-        );
-      }
+  const filteredTokenList = useMemo(() => {
+    const filteredTokens: (
```

Comment:
If I'm not wrong you can return an early return if `searchQuery` is empty:
`if (!searchQuery.trim()) return [];`

Reply:
If searchQuery is empty we don't apply a symbol filter and need to return all tokens. I modified the condition to skip checking for a symbol match when searchQuery is empty instead